tests/Unit/EnclosureTest.php

```diff
@@ -58,4 +60,69 @@ public function testClassExists()
     {
         $this->assertTrue(class_exists('SimplePie_Enclosure'));
     }
+
+    /**
+     * @dataProvider getLinkProvider
+     */
+    public function test_get_link($data, $expected)
+    {
+        $feed = new SimplePie();
+        $feed->set_raw_data($data);
+        $feed->enable_cache(false);
+        $feed->init();
+
+        $item = $feed->get_item(0);
+        $enclosure = $item->get_enclosure(0);
+
+        $this->assertInstanceOf(Enclosure::class, $enclosure);
+        $this->assertSame($expected, $enclosure->get_link());
+    }
+
+    public function getLinkProvider()
+    {
+        return [
+            'Test enclosure get_link urlencoded' => [
+<<<EOT
+<rss version="2.0" xmlns:media="http://search.yahoo.com/mrss/">
+    <channel>
+        <title>Test enclosure link 1</title>
+        <description>Test enclosure link 1</description>
+        <link>http://example.net/tests/</link>
+        <item>
+            <title>Test enclosure link 1.1</title>
+            <description>Test enclosure link 1.1</description>
+            <guid>http://example.net/tests/#1.1</guid>
+            <link>http://example.net/tests/#1.1</link>
+            <media:content url="http://example.net/link?a=%22b%22&amp;c=%3Cd%3E" medium="image">
+            </media:content>
+        </item>
+    </channel>
+</rss>
+EOT
+            ,
+                'http://example.net/link?a=%22b%22&amp;c=%3Cd%3E',
+            ],
+            'Test enclosure get_link urldecoded' => [
+<<<EOT
+<rss version="2.0" xmlns:media="http://search.yahoo.com/mrss/">
+    <channel>
+        <title>Test enclosure link 2</title>
+        <description>Test enclosure link 2</description>
+        <link>http://example.net/tests/</link>
+        <item>
+            <title>Test enclosure link 2.1</title>
+            <description>Test enclosure link 2.1</description>
+            <guid>http://example.net/tests/#2.1</guid>
+            <link>http://example.net/tests/#2.1</link>
+            <media:content url="http://example.net/link?a=&quot;b&quot;&amp;c=&lt;d&gt;" medium="image">
+            </media:content>
+        </item>
+    </channel>
+</rss>
+EOT
+            ,
+                'http://example.net/link?a=%22b%22&amp;c=%3Cd%3E',
+            ],
+        ];
```

Comment:
Might be a bit cleaner to switch to `yield` to reduce unnecessary indentation.

And switching the heredoc mark to `XML` will allow some editors (e.g. Sublime Text) to highlight it as such.

Ideally we would also move the ending mark to proper level but [flexible heredoc syntax](https://wiki.php.net/rfc/flexible_heredoc_nowdoc_syntaxes) is only available since PHP 7.3. Fortunately, this document is not whitespace sensitive so the extra indentation will not matter. We can move the marker once we bump the minimal PHP version.

```suggestion
        yield 'Test enclosure get_link urlencoded' => [
            <<<XML
            <rss version="2.0" xmlns:media="http://search.yahoo.com/mrss/">
                <channel>
                    <title>Test enclosure link 1</title>
                    <description>Test enclosure link 1</description>
                    <link>http://example.net/tests/</link>
                    <item>
                        <title>Test enclosure link 1.1</title>
                        <description>Test enclosure link 1.1</description>
                        <guid>http://example.net/tests/#1.1</guid>
                        <link>http://example.net/tests/#1.1</link>
                        <media:content url="http://example.net/link?a=%22b%22&amp;c=%3Cd%3E" medium="image">
                        </media:content>
                    </item>
                </channel>
            </rss>
XML
            ,
            'http://example.net/link?a=%22b%22&amp;c=%3Cd%3E',
        ];

        yield 'Test enclosure get_link urldecoded' => [
            <<<XML
            <rss version="2.0" xmlns:media="http://search.yahoo.com/mrss/">
                <channel>
                    <title>Test enclosure link 2</title>
                    <description>Test enclosure link 2</description>
                    <link>http://example.net/tests/</link>
                    <item>
                        <title>Test enclosure link 2.1</title>
                        <description>Test enclosure link 2.1</description>
                        <guid>http://example.net/tests/#2.1</guid>
                        <link>http://example.net/tests/#2.1</link>
                        <media:content url="http://example.net/link?a=&quot;b&quot;&amp;c=&lt;d&gt;" medium="image">
                        </media:content>
                    </item>
                </channel>
            </rss>
XML
            ,
            'http://example.net/link?a=%22b%22&amp;c=%3Cd%3E',
        ];
```

Reply:
I had kept the same syntax than other tests, so for consistency, it could be an idea to update the rest.

Reply:
If someone wants to do the work, then we can do that. On the other hand, we could also keep the old syntax for consistency and let php-cs-fixer fix it automatically once we have increased the PHP version.